src/content_mapblock.cpp

```diff
@@ -280,45 +282,61 @@ void MapblockMeshGenerator::drawCuboid(const aabb3f &box,
 // Gets the base lighting values for a node
 void MapblockMeshGenerator::getSmoothLightFrame()
 {
+	for (int k = 0; k < 8; ++k)
+		frame.sunlight[k] = false;
 	for (int k = 0; k < 8; ++k) {
 		LightPair light(getSmoothLightTransparent(blockpos_nodes + p, light_dirs[k], data));
-		frame.lightsA[k] = light.lightA;
-		frame.lightsB[k] = light.lightB;
+		frame.lightsDay[k] = light.lightDay;
+		frame.lightsNight[k] = light.lightNight;
+		// If there is direct sunlight and no ambient occlusion at some corner,
+		// mark the vertical edge (top and bottom corners) containing it.
+		if (light.lightDay == 255) {
+			frame.sunlight[k] = true;
+			frame.sunlight[k ^ 2] = true;
```

Comment:
I'm just wondering how k ^ 2 will get the bottom corner vertex number here from the top corner number?

Reply:
Nevermind, '^' is XOR not power.